static/app/components/integrationExternalMappings.tsx

```diff
@@ -3,39 +3,109 @@ import styled from '@emotion/styled';
 import capitalize from 'lodash/capitalize';
 
 import Access from 'sentry/components/acl/access';
+import MenuItemActionLink from 'sentry/components/actions/menuItemActionLink';
 import Button from 'sentry/components/button';
-import Confirm from 'sentry/components/confirm';
+import DropdownLink from 'sentry/components/dropdownLink';
+import IntegrationExternalMappingForm from 'sentry/components/integrationExternalMappingForm';
 import Pagination from 'sentry/components/pagination';
 import {Panel, PanelBody, PanelHeader, PanelItem} from 'sentry/components/panels';
 import Tooltip from 'sentry/components/tooltip';
-import {IconAdd, IconDelete, IconEdit} from 'sentry/icons';
+import {IconAdd, IconArrow, IconEllipsis} from 'sentry/icons';
 import {t, tct} from 'sentry/locale';
+import PluginIcon from 'sentry/plugins/components/pluginIcon';
 import space from 'sentry/styles/space';
-import {ExternalActorMapping, Integration} from 'sentry/types';
+import {ExternalActorMapping, Integration, Organization} from 'sentry/types';
 import {getIntegrationIcon} from 'sentry/utils/integrationUtil';
 import EmptyMessage from 'sentry/views/settings/components/emptyMessage';
 
-type Props = {
+type Props = Pick<
+  IntegrationExternalMappingForm['props'],
+  'dataEndpoint' | 'getBaseFormEndpoint' | 'sentryNamesMapper' | 'onResults'
+> & {
+  organization: Organization;
   integration: Integration;
-  mappings: {id: string; externalName: string; sentryName: string}[];
+  mappings: ExternalActorMapping[];
   type: 'team' | 'user';
-  onCreateOrEdit: (mapping?: ExternalActorMapping) => void;
+  onCreate: (mapping?: ExternalActorMapping) => void;
   onDelete: (mapping: ExternalActorMapping) => void;
   pageLinks?: string;
 };
 
 type State = {};
-
 class IntegrationExternalMappings extends Component<Props, State> {
-  render() {
-    const {integration, mappings, type, onCreateOrEdit, onDelete, pageLinks} = this.props;
+  renderMappingName(mapping: ExternalActorMapping, hasAccess: boolean) {
+    const {
+      type,
+      getBaseFormEndpoint,
+      integration,
+      dataEndpoint,
+      sentryNamesMapper,
+      onResults,
+    } = this.props;
+    const mappingName = mapping.sentryName ?? '';
+    return hasAccess ? (
+      <IntegrationExternalMappingForm
+        type={type}
+        integration={integration}
+        dataEndpoint={dataEndpoint}
+        getBaseFormEndpoint={getBaseFormEndpoint}
+        mapping={mapping}
+        sentryNamesMapper={sentryNamesMapper}
+        onResults={onResults}
+        isInline
+      />
+    ) : (
+      mappingName
+    );
+  }
 
+  renderMappingOptions(mapping: ExternalActorMapping, hasAccess: boolean) {
```

Comment:
This is just to have a more readable component. This chooses to render the dropdown with delete, or tooltip. Eventually this can be used to house the 'ignore' option for suggestions, but that's not the scope of this PR.